src/mercury_engine_data_structures/version_validation.py

```diff
@@ -0,0 +1,62 @@
+from hashlib import md5
+
+from construct.core import BytesInteger, Int32ul, PrefixedArray, Struct
+
+from mercury_engine_data_structures.file_tree_editor import FileTreeEditor, GameVersion
```

Comment:
Add `from __future__ import annotations` to the file, then the FileTreeEditor can be in a TYPE_CHECKING block which solves the import loop.